src/compiler/utilities.ts

```diff
@@ -5511,10 +5511,9 @@ namespace ts {
     }
 
     /** Add a value to a set, and return true if it wasn't already present. */
-    export function addToSeen(seen: ESMap<string, true>, key: string | number): boolean;
-    export function addToSeen<T>(seen: ESMap<string, T>, key: string | number, value: T): boolean;
-    export function addToSeen<T>(seen: ESMap<string, T>, key: string | number, value: T = true as any): boolean {
-        key = String(key);

```

Comment:
I wanted to use `addToSeen` with a `Map<number, Something>`, and this coercion prevented me from doing that. It doesn’t seem to make a lot of sense now that we have proper ES Maps that can take keys of any type. But I can revert this and the cascading small updates if there’s any concern about it.